### PR TITLE
oauth2MetadataUrl

### PR DESCRIPTION
**Pull request by AxelNennker**
Add oauth2MetadataUrl to oauth2 allows the client to download the OAuth2 metadata to download the RFC8414 OAuth2 metadata.

Fixes https://github.com/OAI/OpenAPI-Specification/issues/3595

### CODE REVIEW

**Comment by LasneF** [comment]
Looks fair

**Comment by AxelNennker** [comment]
Like this?

```
diff --git a/schemas/v3.1/schema.yaml b/schemas/v3.1/schema.yaml
index 68fcc0d..298ff51 100644
--- a/schemas/v3.1/schema.yaml
+++ b/schemas/v3.1/schema.yaml
@@ -799,6 +799,9 @@ $defs:
           properties:
             flows:
               $ref: '#/$defs/oauth-flows'
+            oauth2MetadataUrl:
+              type: string
+              format: uri
           required:
             - flows

```

**Comment by lornajane** [comment]
The meta schema can be updated later on, it doesn't need to be part of this change.

File: versions/3.2.0.md [hunk 1]
```diff
@@ -3199,6 +3199,7 @@ Field Name | Type | Applies To | Description
 <a name="securitySchemeBearerFormat"></a>bearerFormat | `string` | `http` (`"bearer"`) | A hint to the client to identify how the bearer token is formatted.  Bearer tokens are usually generated by an authorization server, so this information is primarily for documentation purposes.
 <a name="securitySchemeFlows"></a>flows | [OAuth Flows Object](#oauthFlowsObject) | `oauth2` | **REQUIRED**. An object containing configuration information for the flow types supported.
 <a name="securitySchemeOpenIdConnectUrl"></a>openIdConnectUrl | `string` | `openIdConnect` | **REQUIRED**. OpenId Connect URL to discover OAuth2 configuration values. This MUST be in the form of a URL. The OpenID Connect standard requires the use of TLS.
+<a name="securitySchemeoauth2MetadataUrl"></a>oauth2MetadataUrl | `string` | `oauth2` | **OPTIONAL**. A HTTPS URL to discover OAuth2 configuration values. This must be in the form of a URL. See [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
```
Comment (lornajane) [inline versions/3.2.0.md:3202]:
```suggestion
<a name="securitySchemeoauth2MetadataUrl"></a>oauth2MetadataUrl | `string` | `oauth2` |A HTTPS URL to discover OAuth2 configuration values. This must be in the form of a URL. See [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
```

The "OPTIONAL" notation seems out of place here. We indicate which are the required fields, but I don't think we usually indicate which are optional.

**Comment by AxelNennker** [comment]
Hi, the longer this mini PR stays unresolved the bigger the mess I create. Sorry.
It seems I did something that led to many more commits being in this.
Could somebody please help me getting this straight again, please?
Otherwise, I would go the radical way and close this PR, delete my fork, re-create it and create a new PR. But I guess there is a more elegant way to add this mini PR to OAI.

**Comment by lornajane** [comment]
@AxelNennker I'l take a look and see if I can untangle it!

**Comment by lornajane** [comment]
I can't guess what happened here, but I don't see your original commits anywhere so I can't recover them.

As you can tell from our pull request queue, it's not unusual for pull requests to be open for longer periods of time, so it would be good to understand what happened here so you can contribute this change (and hopefully many others) safely! The fastest route forward is to recreate the pull request (I'm sorry, I realise this wasn't the first time you proposed this change, but I'm in support of us accepting it so let's persevere). My tips for success:
 - branch from the 3.2.0 branch.
 - open the pull request from that branch.
 - don't click any "update" buttons presented to you by GitHub, it has a very limited view of how git should be used and it doesn't fit our workflow. If your PR gets so out of date that it needs to be rebased, we can help you to do that,

I'm closing this pull request because I feel like I have to - we can't merge it as it is - but I'm looking out for it's replacement!

**Pull request closed without merging**